### PR TITLE
Stage 5 & 6 (Clear History function)

### PR DESCRIPTION
**Pull request by n8han11**
Added Clear History function for Stages 5 & 6.

### CODE REVIEW

File: Stages/history.c [hunk 1]
```diff
@@ -141,4 +138,27 @@ int load_history(char* directory)
...
+    // clear history array
+    for (int i = 0; i < HISTORY_SIZE; i++) {
+        history[i].number = 0;
+        memset(history[i].command, 0, MAX_LINE_LENGTH);
+    }
+
+    // wipe history file
+    char fileLoc[2048];
+    sprintf(fileLoc, "%s/%s", directory, HIST_FILE_NAME);
+    FILE *fp = fopen(fileLoc, "w");
```
Comment (mohfez) [inline Stages/history.c:156]: Can use set_file() from file_handling.c here, would look cleaner.

File: Stages/history.c [hunk 2]
```diff
@@ -141,4 +138,27 @@ int load_history(char* directory)
...
+    if (fp == NULL) {
+        printf("Error: Failed to open history file for writing.\n");
+        return;
+    }
+    fclose(fp);
+
+    printf("History cleared.\n");
```
Comment (mohfez) [inline Stages/history.c:163]: make it return an int to indicate success/failure then handle the print messages in shell.c based on the int returned.

(you can maybe do `return set_file(...);`)

File: shell.c [hunk 3]
```diff
@@ -197,6 +197,12 @@ int main()
 					printf("Unalias can only accept exactly one argument.\n");
 				}
 			}
+
+			else if (strcmp("clear", args[0]) == 0 && strcmp("history", args[1]) == 0)
```
Comment (mohfez) [inline shell.c:201]: use argsLen to check if it's ONLY 2 arguments; no more, no less.
if it's not then add appropriate print error messages.

File: shell.c [hunk 4]
```diff
@@ -197,6 +197,12 @@ int main()
...
+            		{
+                		clear_history(home_dir);
```
Comment (mohfez) [inline shell.c:203]: use the int that it returns to print the success/failure message as discussed earlier.

**Review by mohfez — CHANGES_REQUESTED** [review]

**Comment by mohfez** [comment]
I made the changes in the other branch.

Thanks.

**Comment by VictorVictini** [comment]
np

**Pull request closed without merging**